NotesBackEnd/src/Notes.Api/Controllers/NotesController.cs

```diff
@@ -2,37 +2,59 @@
 using System.Threading.Tasks;
 using System.Web.Http;
 using Microsoft.Web.Http;
-using Notes.Api.Model;
+using Notes.Contracts.ApiServices;
+using Notes.Contracts.Model;
+using Notes.Contracts.Repository;
 
 namespace Notes.Api.Controllers
 {
-
     [ApiVersion("1.0")]
     [Route("api/v{version:apiVersion}/notes/{id?}", Name = NotesRouteName)]
     public class NotesController : ApiController
     {
         internal const string NotesRouteName = "Notes";
+        private readonly INotesRepository _repository;
+        private readonly IUrlLocationHelper _locationHelper;
 
-        private static readonly Note[] Notes = {
-            new Note {Text = "First note", Id = new Guid("2c00d1c2-fd2b-4c06-8f2d-130e88f719c2")},
-            new Note {Text = "Second note", Id = new Guid("ebcb3d81-af4e-428f-a22d-e7852d70d3a0")},
-            new Note {Text = "Third note", Id = new Guid("599442c0-ae28-4157-9a3f-0491bb4ba6c1")},
-            new Note {Text = "Fourth note", Id = new Guid("4785546e-824d-42a4-900b-e656f19ffb59")}
-        };
+        public NotesController(INotesRepository repository, IUrlLocationHelper locationHelper)
+        {
+            _repository = repository;
+            _locationHelper = locationHelper;
+        }
 
         public async Task<IHttpActionResult> GetAsync()
-            => await Task.FromResult(Ok(Notes));
+        {
+            var notes = await _repository.GetAllNotesAsync();
+
+            return await Task.FromResult(Ok(notes));

```

Comment:
No need to `await` here. By making the method `async` you automatically return `Task` with anything you actually return. You only need `await` at the repository call since it's another async method. Here, you can just `return Ok(notes);`